var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -70,14 +69,18 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('hiop@0.5.3:', when='@1.3.0:+hiop')
 
     depends_on('hiop+cuda', when='+hiop+cuda')
-    depends_on('hiop+rocm', when='+hiop+rocm')
     depends_on('hiop~mpi', when='+hiop~mpi')
     depends_on('hiop+mpi', when='+hiop+mpi')
 
     depends_on('petsc@3.13:3.14', when='@:1.2.99')
     depends_on('petsc@3.16.0:3.16', when='@1.3.0:')
     depends_on('petsc~mpi', when='~mpi')
 
+    for arch in ROCmPackage.amdgpu_targets:
+        rocm_dep = "+rocm amdgpu_target={0}".format(arch)
+        depends_on("hiop {0}".format(rocm_dep), when=rocm_dep)
+        depends_on("raja {0}".format(rocm_dep), when="+raja {0}".format(rocm_dep))
+
```

Comment:
I've had difficulty propogating CUDA architectures to some dependencies for other spack based packages that I have installed - would a similar patch apply to those packages as well to maintain consistent CUDA arch across a concretization?

Reply:
Yes definitely, I just pushed https://github.com/spack/spack/pull/30259/commits/2a5d28ab8753c59c940397d23d0b2f6fa4e0bc29 which uses the same style logic to handle passing of `cuda_arch` value to dependencies when `+cuda`

Reply:
Maybe I should update https://github.com/spack/spack/pull/30184 so that it also includes logic for passing `cuda_arch` to dependencies in the `+cuda` case? What do you think?

Reply:
It seems as though the HiOp changes have already been merged - perhaps you can add that to this PR

Reply:
See https://github.com/spack/spack/pull/30259/commits/4c669eabeb987545800b6d9130fa65606be81ed9